benchmarks/nemo/nemolite2d/psykal/README.md

```diff
@@ -12,36 +12,63 @@ to set things up for Gnu:
+the same parameters.
+
 Since the Makefile uses PSyclone you must have PSyclone installed on your
 system (see http://psyclone.readthedocs.io/en/stable/getting_going.html).
 
-The Makefile supports two targets:
+The Makefile supports many targets:
+
+- nemolite2d_serial (default) - vanilla (serial) version
+- nemolite2d_omp - OpenMP version with static scheduling
+- nemolite2d_mpi - MPI version for distributed memory parallelism
+- nemolite2d_hybrid - MPI and OpenMP combined version
+- nemolite2d_acc - OpenACC (Not working - PSyclone will not generate it)
+- nemolite2d_ocl - OpenCL version
+- nemolite2_mpiocl - MPI and OpenCL combined version (Not working - Psyclone
+will generate a OpenCL-only version)
 
-1. nemolite2d_gen - vanilla (serial) version
-2. nemolite2d_gen_omp - OpenMP version with static scheduling
+Executing `make <target>` will generate a folder with the same name of the
+target, containing at least a `alg.f90` and a `psy.f90` for the
+PSyclone-generated Algorithm and PSy-layer components respectively.
+It will also generate kernel source files when necessary. Finally, it will
+build a `nemolite2d.exe` with the source files in that target folder.
 
-but just executing `make` will build both and should result in the
-creation of two executables - `nemolite2d_gen.exe` and
-`nemolite2d_gen_omp.exe`. (The PSyclone-generated PSy code for these
-two binaries can be found in `psy.f90` and `psy_omp.f90`, respectively.)
 
 ## Running ##
 
 Model parameters (size of domain [jpiglo,jpjglo], number of time-steps
 [nitend], whether or not and how often to do output [irecord]) may be
 configured by editing the `namelist` file.
 
-Since `nemolite2d_gen.exe` is serial, it can just be executed on
-the command line, e.g.:
-
-    > ./nemolite2d_gen.exe
-
-For the OpenMP version, you will need to set OMP_NUM_THREADS to the
-desired number of threads to use.
-
-Note that if you are benchmarking performance then you probably need
-to ensure that the threads are bound sensibly to processor cores (see
-e.g. taskset, likwid or Intel's KMP_AFFINITY).
+There are differences on how to execute the generated `nemolite2d_gen.exe`,
+notably:
+
+- For the OpenMP version, you will need to set `OMP_NUM_THREADS` to the
+desired number of threads to use. Note that if you are benchmarking
+performance then you probably need to ensure that the threads are bound
+sensibly to processor cores (see e.g. taskset, likwid or Intel's KMP_AFFINITY).
+
+- For the MPI version, you will need to prefix the binary with the
+`mpirun -n <#ranks>` command. See additional MPI parameters for the selected
+implementation with `mpirun -h`.
+
+- For the OpenCL version, you will need to specify which kernels file to use
+with the `FORTCL_KERNELS_FILE` environment variable, it can be a source file
+for JITed OpenCL or a ahead-of-time compiled binary object. Optionally you can
+select with OpenCL platform to use with the `FORTCL_PLATFORM` environment
```

Comment:
s/with OpenCL/which OpenCL/ and please add a pointer to the FortCL 'documentation' at https://github.com/stfc/FortCL

Reply:
Done